custom_components/heatmiserneo/climate.py

```diff
@@ -164,76 +137,55 @@ def hvac_modes(self):
         """Return the list of available operation modes."""
         return self._hvac_modes
 
-    # @property
-    # def preset_mode(self):
-    #     """Return preset mode."""
-    #     return self._preset
-
-    # @property
-    # def preset_modes(self):
-    #     """Return preset modes."""
-    #     return self._preset_modes
-
-    def set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs):
         """ Set new target temperature. """
-        response = self._hub.json_request({"SET_TEMP": [int(kwargs.get(ATTR_TEMPERATURE)), self._name]})
+        response = await self._hub.set_temp(
+            int(kwargs.get(ATTR_TEMPERATURE)), [self._neostat]
+        )
         if response:
             _LOGGER.info("set_temperature response: %s " % response)
             # Need check for success here
             # {'result': 'temperature was set'}
 
-    def set_hvac_mode(self, hvac_mode):
+    async def async_set_hvac_mode(self, hvac_mode):
         """Set hvac mode."""
+        frost: bool
+
         if hvac_mode == HVAC_MODE_HEAT:
             self._hvac_mode = HVAC_MODE_HEAT
-            mode = "FROST_OFF"
+            frost = False
         elif hvac_mode == HVAC_MODE_OFF:
             self._hvac_mode = HVAC_MODE_OFF
-            mode = "FROST_ON"
+            frost = True
         else:
             _LOGGER.error("Unrecognized hvac mode: %s", hvac_mode)
             return
 
-        response = self._hub.json_request({mode: [self._name]})
+        response = await self._hub.frost(frost, [self._neostat])
```

Comment:
Was there any particular reason not to use `await self._neostat.frost(frost)` ? Seems slightly shorter.